extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/SplitPackageProcessor.java

```diff
@@ -94,7 +94,8 @@ void splitPackageDetection(ApplicationArchivesBuildItem archivesBuildItem,
                             }
                         }
                     } else {
-                        splitPackages.add(a.getGroupId() + ":" + a.getArtifactId());
+                        splitPackages.add(a.getGroupId() + ":" + a.getArtifactId() + ":" + a.getType() +
+                                (a.getClassifier().isEmpty() ? "" : (":" + a.getClassifier())));
```

Comment:
Just in case, if you want to be compatible with the `AppArtifactKey.toString()` the format should be `g:a:c:t`.

Reply:
@aloubyansky good point, thanks. I have fixed that.